nat/behavior/discovery.go

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright (C) 2021 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package behavior
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/pion/stun"
+)
+
+// Enum of DiscoverNATBehavior return values
+const (
+	NatTypeNone               = "none"
+	NatTypeFullCone           = "fullcone"
+	NatTypeRestrictedCone     = "rcone"
+	NatTypePortRestrictedCone = "prcone"
+	NatTypeSymmetric          = "symmetric"
+)
+
+// Enum of DiscoverNATMapping return values
+const (
+	MappingNone                 = "none"
+	MappingIndependent          = "independent"
+	MappingAddressDependent     = "address"
+	MappingAddressPortDependent = "addressport"
+)
+
+// Enum of DiscoverNATFiltering return values
+const (
+	FilteringIndependent = "independent"
+	FilteringAddress     = "address"
+	FilteringAddressPort = "addressport"
+)
+
+// DEFAULT_TIMEOUT for each single STUN request
+const DefaultTimeout = 3 * time.Second
+
+type stunServerConn struct {
+	conn        net.PacketConn
+	LocalAddr   net.Addr
+	RemoteAddr  *net.UDPAddr
+	OtherAddr   *net.UDPAddr
+	messageChan chan *stun.Message
+}
+
+func (c *stunServerConn) Close() error {
+	return c.conn.Close()
+}
+
+// STUN protocol compatibility errors
```

Comment:
I'd put these vars next to the const above, so it will be easy to find them.

Reply:
Updated